videoprocess/vpphdr_tm.cpp

```diff
@@ -0,0 +1,861 @@
+
+    return va_status;
+}
+
+static VAStatus
+hdrtm_filter_init(VABufferID *filter_param_buf_id, uint32_t tm_type)
+{
+    VAStatus va_status = VA_STATUS_SUCCESS;
+    VAProcFilterParameterBufferHDRToneMapping hdrtm_param;
+
+    VAHdrMetaDataHDR10 in_hdr10_metadata = {};
+
+	// The input is HDR content
+    in_hdr10_metadata.max_display_mastering_luminance = g_in_max_display_luminance;
+    in_hdr10_metadata.min_display_mastering_luminance = g_in_min_display_luminance;
+    in_hdr10_metadata.max_content_light_level         = g_in_max_content_luminance;
+    in_hdr10_metadata.max_pic_average_light_level     = g_in_pic_average_luminance;
+    in_hdr10_metadata.display_primaries_x[0] = 8500;
+    in_hdr10_metadata.display_primaries_y[0] = 39850;
+    in_hdr10_metadata.display_primaries_x[1] = 35400;
+    in_hdr10_metadata.display_primaries_y[1] = 14600;
+    in_hdr10_metadata.display_primaries_x[2] = 6550;
+    in_hdr10_metadata.display_primaries_y[2] = 2300;
+    in_hdr10_metadata.white_point_x = 15635;
+    in_hdr10_metadata.white_point_y = 16450; 
+
+    hdrtm_param.type = VAProcFilterHighDynamicRangeToneMapping;
+    hdrtm_param.data.metadata_type = VAProcHighDynamicRangeMetadataHDR10;
+    hdrtm_param.data.metadata= &in_hdr10_metadata;
+    hdrtm_param.data.metadata_size = sizeof(VAHdrMetaDataHDR10);   
+	
+    va_status = vaCreateBuffer(va_dpy, context_id, VAProcFilterParameterBufferType, sizeof(hdrtm_param), 1, (void *)&hdrtm_param, filter_param_buf_id);
+
+    return va_status;
+}
+
+static VAStatus
+hdrtm_metadata_init(VAHdrMetaData &out_metadata, uint32_t tm_type)
+{
+    VAStatus va_status = VA_STATUS_SUCCESS;
+
+    VAHdrMetaDataHDR10 out_hdr10_metadata = {}; 
+
+    out_hdr10_metadata.max_display_mastering_luminance = g_out_max_display_luminance;
+    out_hdr10_metadata.min_display_mastering_luminance = g_out_min_display_luminance;
+    out_hdr10_metadata.max_content_light_level         = g_out_max_content_luminance;
+    out_hdr10_metadata.max_pic_average_light_level     = g_out_pic_average_luminance;
+    printf("hdrtm_metadata_init g_out_max_display_luminance %d, g_out_min_display_luminance %d\n", g_out_max_display_luminance, g_out_min_display_luminance);
+    printf("hdrtm_metadata_init g_out_max_content_luminance %d, g_out_pic_average_luminance %d\n", g_out_max_content_luminance, g_out_pic_average_luminance);
+
+    // HDR display or SDR display
+    switch (tm_type)
+    {
+    case VA_TONE_MAPPING_HDR_TO_HDR:
+        out_hdr10_metadata.display_primaries_x[0] = 8500;
+        out_hdr10_metadata.display_primaries_y[0] = 39850;
+        out_hdr10_metadata.display_primaries_x[1] = 35400;
+        out_hdr10_metadata.display_primaries_y[1] = 14600;
+        out_hdr10_metadata.display_primaries_x[2] = 6550;
+        out_hdr10_metadata.display_primaries_y[2] = 2300;
+        out_hdr10_metadata.white_point_x = 15635;
+        out_hdr10_metadata.white_point_y = 16450;
+        break;
```

Comment:
hdr to hdr case,the meta data will keep unchanged?

Reply:
In the example, just change the luminance. Color primaries unchanged.